### PR TITLE
Added Knapcask exercise

### PR DESCRIPTION
**Pull request by tofische**
Added a new Haskell practice exercise Knapsack based on the available problem specification.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1109,6 +1120,19 @@
...
+        "name": "Rational Numbers",
+        "uuid": "b4426fe9-2e49-48c5-9dea-c86032009687",
+        "practices": [],
+        "prerequisites": [],
+        "difficulty": 3,
+        "topics": [
+          "define_type",
+          "math",
+          "number_theory"
+        ]
```
Comment (ErikSchierboom) [inline config.json:1134]: This seems to be unrelated

File: exercises/practice/knapsack/.meta/config.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "authors": [],
```
Comment (ErikSchierboom) [inline exercises/practice/knapsack/.meta/config.json:2]: Could you add your github username to this array?

File: exercises/practice/leap/.docs/instructions.md
Comment (ErikSchierboom) [inline exercises/practice/leap/.docs/instructions.md:1]: This seems to be an unrelated change. Could you remove all unrelated changes from the PR?

**Review by ErikSchierboom — APPROVED** [review]
Looks great!

**Comment by MatthijsBlom** [comment]
Sorry for being late.

I would have preferred `data Item = Item { weight :: Int, value :: Int }`.
